File: android/app/src/main/java/app/unimusic/CleanWebViewClient.java
package app.unimusic;

import android.webkit.WebResourceRequest;
import android.webkit.WebResourceResponse;
import android.webkit.WebView;

import com.getcapacitor.Bridge;
import com.getcapacitor.BridgeWebViewClient;

/**
 * WebView Client which strips request of the HTTP Client Hint headers
 * Which are used (mostly) by Google to identify the users platform and its features
 *
 * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Client_hints" />
 */
public class CleanWebViewClient extends BridgeWebViewClient {
    public CleanWebViewClient(Bridge bridge) {
        super(bridge);
    }

    private void stripClientHintsHeaders(WebResourceRequest request) {
        var headers = request.getRequestHeaders();
        var iterator = headers.entrySet().iterator();
        while (iterator.hasNext()) {
            String key = iterator.next().getKey();
            if (
                    key.equalsIgnoreCase("Referer") ||
                            key.equalsIgnoreCase("Origin") ||
                            key.equalsIgnoreCase("Sec-CH-UA")
                            || key.equalsIgnoreCase("Sec-CH-UA-Mobile")
                            || key.equalsIgnoreCase("Sec-CH-UA-Platform")
            ) {
                iterator.remove();
            }
        }
    }

    @Override
    public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
        stripClientHintsHeaders(request);
        return super.shouldOverrideUrlLoading(view, request);
    }

    @Override
    public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceRequest request) {
        stripClientHintsHeaders(request);
        return super.shouldInterceptRequest(view, request);
    }
}

File: android/app/src/main/java/app/unimusic/CleanWebViewClient.kt
package app.unimusic

import android.webkit.WebResourceRequest
import android.webkit.WebResourceResponse
import android.webkit.WebView
import com.getcapacitor.Bridge
import com.getcapacitor.BridgeWebViewClient

/**
 * WebView Client which strips request of the HTTP Client Hint headers
 * Which are used (mostly) by Google to identify the users platform and its features
 *
 * See: [Client Hints](https://developer.mozilla.org/en-US/docs/Web/HTTP/Client_hints)
 */
class CleanWebViewClient(bridge: Bridge?) : BridgeWebViewClient(bridge) {
    private fun stripClientHintsHeaders(request: WebResourceRequest) {
        val headers = request.requestHeaders
        val iterator = headers.entries.iterator()

        while (iterator.hasNext()) {
            val key: String = iterator.next()!!.key!!;

            if (
                key.equals("Referer", ignoreCase = true) ||
                key.equals("Origin", ignoreCase = true) ||
                key.equals("Sec-CH-UA", ignoreCase = true) ||
                key.equals("Sec-CH-UA-Mobile", ignoreCase = true) ||
                key.equals("Sec-CH-UA-Platform", ignoreCase = true)
            ) {
                iterator.remove()
            }
        }
    }

    override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest): Boolean {
        stripClientHintsHeaders(request)
        return super.shouldOverrideUrlLoading(view, request)
    }

    override fun shouldInterceptRequest(
        view: WebView?,
        request: WebResourceRequest
    ): WebResourceResponse? {
        stripClientHintsHeaders(request)
        return super.shouldInterceptRequest(view, request)
    }
}
